synapse/cores/common.py

```diff
@@ -3705,3 +3709,74 @@ def delBlobValu(self, key):
             NoSuchName: If the key is not present in the store.
         '''
         return self.store.delBlobValu(key)
+
+    def _onSetMembranes(self, membranes):
+        [self._initMembrane(name, rules) for name, rules in membranes]
+
+    # TODO plumb in some way to consume/handle fifo message
+    def _initMembrane(self, name, rules):
+        if self._core_membranes.get(name):
+            raise s_common.MembraneExists()
+
+        def fn(mesg):
+            return self.putCoreFifo(name, mesg)
+
+        membrane = s_membrane.Membrane(name, rules, fn)
+        self._core_membranes[name] = membrane
+
+        with self.getCoreXact() as xact:
+            if not self.getTufoByProp('syn:fifo:name', membrane.name):
+                self.formTufoByProp('syn:fifo', '*', name=membrane.name)
+
+        def _filter_fn(mesg):
```

Comment:
TODO: Add this function to the dict entry for the membrane in order to be able to remove it.